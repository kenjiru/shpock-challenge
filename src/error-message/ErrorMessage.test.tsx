import * as React from "react";
import { HTMLProps } from "react";
import { MuiThemeProvider } from "material-ui/styles";
import { mount, ReactWrapper, shallow } from "enzyme";
import shallowToJson from "enzyme-to-json";

import ThemedErrorMessage, { ErrorMessage, IErrorMessageProps } from "./ErrorMessage";

describe("ErrorMessage", () => {
    const ERROR_TEXT: string = "Error text";

    it("should render correctly", () => {
        const output = shallow(
            <ErrorMessage text={ERROR_TEXT}/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it("should have the correct style if is checked", () => {
        const output: ReactWrapper = mount(
            <MuiThemeProvider>
                <ThemedErrorMessage text={ERROR_TEXT}/>
            </MuiThemeProvider>
        );

        const errorMessageProps: IErrorMessageProps = output.find(ErrorMessage).props();
        const errorTextProps: HTMLProps<HTMLElement> = output.find("." + ErrorMessage.ERROR_TEXT_CLASS).props();

        expect(errorTextProps.style.color).toBe(errorMessageProps.muiTheme.textField.errorColor);
    });
});
