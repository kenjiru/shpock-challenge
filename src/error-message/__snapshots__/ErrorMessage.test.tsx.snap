// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should render correctly 1`] = `
<div
  className="error-message"
>
  <FontIcon
    className="material-icons"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": undefined,
      }
    }
  >
    warning
  </FontIcon>
  <span
    className="error-text"
    style={
      Object {
        "color": undefined,
      }
    }
  >
    Error text
  </span>
</div>
`;
